src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -36,25 +36,52 @@ public override void Initialize(AnalysisContext context)
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
             context.RegisterSyntaxNodeAction(
-                AnalyzeSyntaxNode,
+                AnalyzeSyntaxContext,
                 SyntaxKind.ExpressionStatement,
                 SyntaxKind.EqualsValueClause,
                 SyntaxKind.Argument,
                 SyntaxKind.ArrowExpressionClause);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeArrayInitializerContext,
+                SyntaxKind.ArrayInitializerExpression);
         }
 
-        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeSyntaxContext(SyntaxNodeAnalysisContext context)
         {
             var parentSyntaxNode = context.Node;
 
             // Find the invocation expression i.e
             // a method/delegate call or a property access
             // or a chain of them
             var invocationExpressionSyntax = parentSyntaxNode
-                .DescendantNodes()
+                .DescendantNodes(IsNotArrayCreationSyntax)
                 .OfType<InvocationExpressionSyntax>()
                 .FirstOrDefault();
 
+            AnalyzeSyntaxNode(invocationExpressionSyntax, context.ReportDiagnostic);
+
+            bool IsNotArrayCreationSyntax(SyntaxNode syntaxNode) => !syntaxNode.IsKind(SyntaxKind.ArrayCreationExpression);
+        }
+
+        private static void AnalyzeArrayInitializerContext(SyntaxNodeAnalysisContext context)
+        {
+            var arrayInitializerSyntaxNode = context.Node;
+
+            // Find only direct child invocation expressions
+            var invocationExpressions = arrayInitializerSyntaxNode
+                .ChildNodes()
+                .OfType<InvocationExpressionSyntax>();
+
+            // Analyze individual invocation expressions
+            foreach (var invocationExpression in invocationExpressions)
+            {
+                AnalyzeSyntaxNode(invocationExpression, context.ReportDiagnostic);
+            }
+        }
+
+        private static void AnalyzeSyntaxNode(InvocationExpressionSyntax invocationExpressionSyntax, Action<Diagnostic> reportDiagnostic)
```

Comment:
Name this something like generic

Reply:
How about [this](https://github.com/ni/csharp-styleguide/pull/65/commits/6d602757ba6eaca7afa6c713e5275c3aa57a3766)?